include/boost/thread/detail/thread.hpp

```diff
@@ -157,7 +157,7 @@ namespace boost
     }
 namespace thread_detail {
 #ifdef BOOST_THREAD_USES_CHRONO
-#if defined(BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC)
+#if defined(BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC) || defined(BOOST_THREAD_PLATFORM_WIN32)
```

Comment:
Why do you think that WIN32 has a monotonic clock?

Reply:
https://github.com/boostorg/chrono/blob/develop/include/boost/chrono/config.hpp#L68

Reply:
Yes, it has a steady clock, but the WIN32 thread API Boost.Thread is using is a `system_clock` IIUC. I don't know Windows enough. Maybe there is an API that waits on a monotonic clock. 

So on windows, the `internal_clock_t` should be `system_clock` if I'm not wrong.

Reply:
The same for MACOS. It has a steady_clock, but don't provide thread functions with MONOTONIC clocks, AFAIK.

Please, let me know if I'm missing something important.

Reply:
I agree that MAC OS, even though it has a steady clock, doesn't support a steady clock with pthread_cond_init and so the internal clock for MAC OS needs to be system_clock.

For Windows, however, it appears that all *wait_until() functions are internally converted to a duration. So using steady_clock on Windows as the internal clock should work just fine. And actually, because it converts everything to a duration, I don't think any Windows code should be using our internal clock typedefs or implementations at all. All Windows code should just be converting straight from a timepoint to a duration using the timepoint's now() function.

Reply:
You are completely right. I don't know why, I believed we where using the system time.

There are some system_clock uses in win32/basic_timed_mutex.hpp
```
            template <class Rep, class Period>
            bool try_lock_for(const chrono::duration<Rep, Period>& rel_time)
            {
              return try_lock_until(chrono::steady_clock::now() + rel_time);
            }
            template <class Clock, class Duration>
            bool try_lock_until(const chrono::time_point<Clock, Duration>& t)
            {
              using namespace chrono;
              system_clock::time_point     s_now = system_clock::now();
              typename Clock::time_point  c_now = Clock::now();
              return try_lock_until(s_now + ceil<system_clock::duration>(t - c_now));
            }
            template <class Duration>
            bool try_lock_until(const chrono::time_point<chrono::system_clock, Duration>& t)
            {
              using namespace chrono;
              typedef time_point<chrono::system_clock, chrono::system_clock::duration> sys_tmpt;
              return try_lock_until(sys_tmpt(chrono::ceil<chrono::system_clock::duration>(t.time_since_epoch())));
            }
            bool try_lock_until(const chrono::time_point<chrono::system_clock, chrono::system_clock::duration>& tp)
```

My bad, surely :(

However I see that it was already the case for the DAtTime interface

```
            bool timed_lock(::boost::system_time const& wait_until)
            {
                if(try_lock())
                {
                    return true;
                }
                long old_count=active_count;
                mark_waiting_and_try_lock(old_count);

                if(old_count&lock_flag_value)
                {
                    bool lock_acquired=false;
                    void* const sem=get_event();

                    do
                    {
                        if(win32::WaitForSingleObjectEx(sem,::boost::detail::get_milliseconds_until(wait_until),0)!=0)
                        {
                            BOOST_INTERLOCKED_DECREMENT(&active_count);
                            return false;
                        }
                        clear_waiting_and_try_lock(old_count);
                        lock_acquired=!(old_count&lock_flag_value);
                    }
                    while(!lock_acquired);
                }
                return true;
            }

            template<typename Duration>
            bool timed_lock(Duration const& timeout)
            {
                return timed_lock(get_system_time()+timeout);
            }
```

I'll try to see how to fix this.